addons/mass_mailing_sms/models/mailing_trace.py

```diff
@@ -38,6 +38,17 @@ class MailingTrace(models.Model):
         ('sms_optout', 'Opted Out'),
     ])
 
+    @api.depends('sms_sms_id')
+    def _compute_mail_message_id(self):
+        super()._compute_mail_message_id()
+        # traces with emails, smses and snailmail are mutually exclusive sets
+        for trace in self:
+            if not trace.mail_message_id:
+                continue
+            sudo_sms_id = trace.sudo().sms_sms_id
+            if sudo_sms_id:
+                trace.mail_message_id = sudo_sms_id.mail_message_id.sudo(flag=False)
```

Comment:
I think it can be simplified a bit if we filter on `trace_type`
```suggestion
    @api.depends('sms_sms_id')
    def _compute_mail_message_id(self):
        # traces with emails, smses and snailmail are mutually exclusive sets
        sms_trace = self.filtered(lambda trace: trace.trace_type == 'sms')
        super(MailingTrace, self - sms_trace)._compute_mail_message_id()

        for trace in sms_trace:
            trace.mail_message_id = trace.sms_sms_id.mail_message_id
```